Code/Framework/AzCore/AzCore/Debug/Profiler.h

```diff
@@ -51,6 +51,21 @@
 #define AZ_PROFILE_DATAPOINT_PERCENT(...)
 #endif
 
+/**
+ * Macro to declare a profile section for the budget total.
+ * It works similar to AZ_PROFILE_SCOPE but it's meant to be used 
+ * to get budget totals. Using multiple AZ_PROFILE_BUDGET declarations
+ * for the same budget within a callstack will result in incorrect results.
+ */
+#define AZ_PROFILE_BUDGET(budget)                                                                                                          \
+    ::AZ::Debug::ProfileScope AZ_JOIN(azProfileScope, __LINE__)                                                                            \
+    {                                                                                                                                      \
+        AZ_BUDGET_GETTER(budget)(), true, nullptr                                                                                          \
+    }
```

Comment:
Instead of using the ProfileScope class, if we created a separate class/set of functions for this macro, we wouldn't need to have this extra boolean parameter.